dask/dataframe/shuffle.py

```diff
@@ -211,9 +211,15 @@ def set_index(
     upsample: float = 1.0,
     divisions: Sequence | None = None,
     partition_size: float = 128e6,
+    sort: bool = True,
     **kwargs,
 ) -> DataFrame:
     """See _Frame.set_index for docstring"""
+    if not sort:
+        return df.map_partitions(
+            M.set_index, index, align_dataframes=False, drop=drop, **kwargs
+        ).clear_divisions()
```

Comment:
How difficult do you think it would be to add support for `divisions` in here, i.e., validating and setting them?

Reply:
Didn't even consider it, based on https://github.com/dask/dask/issues/10333#issuecomment-1582981517. Should this be in a follow-up you think or I'd be happy to give it a go in this PR.

Reply:
If it's **very** little work, I'd be happy to see it here. If not, I would prefer deferring it to a follow-up issue that we can prioritize accordingly.

Reply:
I think it'd be difficult to know the new divisions ahead of time without a compute

Reply:
As discussed offline with @jrbourbeau, I missed https://github.com/dask/dask/blob/499f4055da707fa76d06a2b79f408f124eee4723/dask/dataframe/shuffle.py#L1101-L1102C1 in https://github.com/dask/dask/pull/10342#discussion_r1227725066, so we can't easily replace `sorted` with `sort`.

Reply:
Okay, I'm assuming this PR satisfies the original issue? Is there anything else you'd like to see added?